ironic/drivers/modules/drac/raid.py

```diff
@@ -644,8 +719,15 @@ def _commit_to_controllers(node, controllers):
         return
 
     driver_internal_info = node.driver_internal_info
+
+    if 'raid_config_substep' not in driver_internal_info:
+        driver_internal_info['raid_config_substep'] = substep
     if 'raid_config_job_ids' not in driver_internal_info:
         driver_internal_info['raid_config_job_ids'] = []
+    if 'raid_config_parameters' not in driver_internal_info:
+        driver_internal_info['raid_config_parameters'] = []
+    if driver_internal_info['raid_config_substep'] == 'complete':
+        del driver_internal_info['raid_config_substep']
```

Comment:
Can you add a command here to delete the raid_config_parameters too?  I'm concerned that if we don't delete it, it will hold data that will affect the execution when the cleaning step is run again.

Reply:
Added  at the end of the function call because it was giving KeyError below while adding job_id